crates/loans/src/lib.rs

```diff
@@ -684,12 +686,9 @@ pub mod pallet {
         #[transactional]
         pub fn activate_market(origin: OriginFor<T>, asset_id: CurrencyId<T>) -> DispatchResultWithPostInfo {
             T::UpdateOrigin::ensure_origin(origin)?;
-            // TODO: if the market is already active throw an error,
-            // to avoid emitting the `ActivatedMarket` event again.
+            // Ensure market state before activating the market
+            Self::check_market_state_before_activating(asset_id)?;
```

Comment:
I had considered modifying `mutate_market`, but since it is also utilized in `update_rate_model`, `update_market`, and `force_update_market` extrinsic, the MarketAlreadyActivated error wouldn't be relevant for those extrinsic as the market can be active and `rate_model` can be changed. Hence, I decided not to alter `mutate_market`.

Reply:
I think you misunderstand me - I am not proposing to check for `MarketAlreadyActivated` inside  `mutate_market` itself. Instead, I propose to change `mutate_market` to accept a closure like this that contains the check:
```rust
Self::mutate_market(asset_id, |stored_market| {
    if let MarketState::Active = stored_market.state {
        return Err(Err(Error::<T>::MarketAlreadyActivated.into())); // this is not allowed currently - needs change to `mutate_market`
    }
    stored_market.state = MarketState::Active;
    Ok(stored_market.clone()) // note the Ok
})?;
```
`